### PR TITLE
Updates GLSL reference to contain GDScript types

### PR DESCRIPTION
**Pull request by KLee1248**
The typing information was ultimately determined in this file where Varients are converted and packed:
https://github.com/godotengine/godot/blob/a51ca2beafc5d74d3e62cb56fbba76ceb39160a3/servers/rendering/renderer_rd/storage_rd/material_storage.cpp#L42

The documentation doesn't contain a reference to dealing with how Packed[Float|Int]Arrays are interpreted, but this should be a much better starting point for new users.

_Closes https://github.com/godotengine/godot-docs/issues/6723_

### CODE REVIEW

**Comment by KLee1248** [comment]
Helps to fix #6723

File: tutorials/shaders/shader_reference/shading_language.rst [hunk 1]
```diff
@@ -788,26 +788,80 @@ GDScript uses different variable types than GLSL does, so when passing variables
...
++----------------------+-------------------------+------------------------------------------------------------+
+| **isampler2D**       | **Texture2D**           |                                                            |
++----------------------+-------------------------+------------------------------------------------------------+
+| **usampler2D**       | **Texture2D**           |                                                            |
++----------------------+-------------------------+------------------------------------------------------------+
+| **sampler2DArray**   | **Array[Texture2D]**    |                                                            |
++----------------------+-------------------------+------------------------------------------------------------+
+| **isampler2DArray**  | **Array[Texture2D]**    |                                                            |
++----------------------+-------------------------+------------------------------------------------------------+
+| **usampler2DArray**  | **Array[Texture2D]**    |                                                            |
```
Comment (clayjohn) [inline tutorials/shaders/shader_reference/shading_language.rst:852]:
```suggestion
| **sampler2DArray**   | **Texture2DArray**    |                                                            |
+----------------------+-------------------------+------------------------------------------------------------+
| **isampler2DArray**  | **Texture2DArray**    |                                                            |
+----------------------+-------------------------+------------------------------------------------------------+
| **usampler2DArray**  | **Texture2DArray**    |                                                            |
```

File: tutorials/shaders/shader_reference/shading_language.rst [hunk 2]
```diff
@@ -788,26 +788,80 @@ GDScript uses different variable types than GLSL does, so when passing variables
...
++----------------------+-------------------------+------------------------------------------------------------+
+| **sampler3D**        | **Texture3D**           |                                                            |
++----------------------+-------------------------+------------------------------------------------------------+
+| **isampler3D**       | **Texture3D**           |                                                            |
++----------------------+-------------------------+------------------------------------------------------------+
+| **usampler3D**       | **Texture3D**           |                                                            |
++----------------------+-------------------------+------------------------------------------------------------+
+| **samplerCube**      | **Cubemap**             |                                                            |
++----------------------+-------------------------+------------------------------------------------------------+
+| **samplerCubeArray** | **Array[Cubemap]**      |                                                            |
```
Comment (clayjohn) [inline tutorials/shaders/shader_reference/shading_language.rst:862]:
```suggestion
| **samplerCubeArray** | **CubemapArray**      |                                                            |
```

File: tutorials/shaders/shader_reference/shading_language.rst [hunk 3]
```diff
@@ -788,26 +788,80 @@ GDScript uses different variable types than GLSL does, so when passing variables
...
++----------------------+-------------------------+------------------------------------------------------------+
+| GLSL type            | GDScript type           | Notes                                                      |
++======================+=========================+============================================================+
+| **bool**             | **bool**                |                                                            |
++----------------------+-------------------------+------------------------------------------------------------+
+| **bvec2**            | **int**                 | Bitwise packed int where bit 0 (LSB) corresponds to x.     |
++----------------------+-------------------------+------------------------------------------------------------+
+| **bvec3**            | **int**                 | Bitwise packed int where bit 0 (LSB) corresponds to x.     |
++----------------------+-------------------------+------------------------------------------------------------+
+| **bvec4**            | **int**                 | Bitwise packed int where bit 0 (LSB) corresponds to x.     |
```
Comment (clayjohn) [inline tutorials/shaders/shader_reference/shading_language.rst:800]: Consider adding an example with GDScript here
i.e. 
For example to set ``bvec2(false, true)`` in the shader using the two bools ``bx = false`` and ``by = true`` in GDScript you would use ``(int(bx) | (int(by) << 1))``

**Review by clayjohn — COMMENTED** [review]
Great work! Just a few suggestions

**Comment by KLee1248** [comment]
Thanks for the review! 

Applied your suggestions and added a GDScript example for bvec2.

**Review by clayjohn — APPROVED** [review]

**Comment by clayjohn** [comment]
Looks great! Thank you

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Thank you, great work! And congrats on your first merged contribution to Godot's docs! 🎉

**Pull request merged**